omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,196 @@
+		if (actionType != TaskAction.Action.RESCHEDULEALLTASKS) {
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException("Cannot execute action " + actionType
+				        + " on empty set of task definitions.");
+			}
+		}
+		
+		switch (action.getAction()) {
+			case SCHEDULETASK:
+				scheduleTasks(taskDefinitions);
+				break;
+			case RESCHEDULETASK:
+				reScheduleTasks(taskDefinitions);
+				break;
+			case DELETE:
+				deleteTasks(taskDefinitions);
+				break;
+			case SHUTDOWNTASK:
+				shutDownTasks(taskDefinitions);
+				break;
+			case RESCHEDULEALLTASKS:
+				reScheduleAllTasks();
+				break;
+		}
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private void scheduleTasks(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.scheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while scheduling task", e);
+			}
+		}
+	}
+	
+	private void shutDownTasks(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.shutDownTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while shutdowning task", e);
+			}
+		}
+	}
+	
+	// Stop and start a set of scheduled tasks.
+	private void reScheduleTasks(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.reScheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while rescheduling task", e);
+			}
+		}
+	}
+	
+	// Stop and start all the tasks.
+	private void reScheduleAllTasks() {
+		try {
+			taskServiceWrapper.reScheduleAllTasks();
+		}
+		catch (SchedulerException e) {
+			throw new APIException("Errors occurred while rescheduling all tasks", e);
+		}
+		
+	}
+	
+	private void deleteTasks(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.deleteTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while deleting task", e);
+			}
+		}
+	}
+	
+	@Override
+	public TaskAction newDelegate() {
+		return new TaskAction();
+	}
+	
+	@Override
+	public TaskAction save(TaskAction delegate) {
+		throw new UnsupportedOperationException("TaskAction cannot be saved");
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("tasks", Representation.REF);
+		description.addProperty("action", "action");
+		return description;
+	}
+	
+	@Override
+	public TaskAction getByUniqueId(String uniqueId) {
+		throw new UnsupportedOperationException("TaskAction cannot get by id");
```

Comment:
Shouldn't we take this to be the internal id, if a number, or task name?

Reply:
Oh sorry. Just ignore the above. 😊

Reply:
Here It is the integer number.  Here I have overridden that method in TaskAction. 
Anyway, TaskAction is not supported to that function.